{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Exploración de los datos"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Downloading cptac_genes.csv: 100%|██████████████████████████████████████████████████| 462k/462k [00:00<00:00, 1.07MB/s]\n",
      "Downloading brca_mapping.csv: 100%|███████████████████████████████████████████████| 6.37k/6.37k [00:00<00:00, 22.5kB/s]\n",
      "Downloading index.tsv: 100%|██████████████████████████████████████████████████████| 30.2k/30.2k [00:00<00:00, 99.0kB/s]\n"
     ]
    }
   ],
   "source": [
    "import cptac"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'brca': 'Breast invasive carcinoma',\n",
       " 'ccrcc': 'Clear cell renal cell carcinoma',\n",
       " 'coad': 'Colon adenocarcinoma',\n",
       " 'gbm': 'Glioblastoma multiforme',\n",
       " 'hnscc': 'Head and Neck squamous cell carcinoma',\n",
       " 'lscc': 'Lung squamous cell carcinoma',\n",
       " 'luad': 'Lung adenocarcinoma',\n",
       " 'ov': 'Ovarian serous cystadenocarcinoma',\n",
       " 'pda': 'Pancreatic ductal adenocarcinoma',\n",
       " 'pdac': 'Pancreatic ductal adenocarcinoma',\n",
       " 'ucec': 'Uterine Corpus Endometrial Carcinoma'}"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cptac.get_cancer_info()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Para este proyecto nos interesan los datos de brca."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>Cancer</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Source</th>\n",
       "      <th>Datatype</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"6\" valign=\"top\">bcm</th>\n",
       "      <th>CNV</th>\n",
       "      <td>[coad, ucec, pdac, ccrcc, ov, lscc, brca, gbm,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>circular_RNA</th>\n",
       "      <td>[ucec, pdac, ccrcc, lscc, gbm, luad, hnscc]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>miRNA</th>\n",
       "      <td>[coad, ucec, pdac, ccrcc, lscc, brca, gbm, lua...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>phosphoproteomics</th>\n",
       "      <td>[coad, ucec, pdac, ccrcc, ov, lscc, brca, gbm,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>proteomics</th>\n",
       "      <td>[coad, ucec, pdac, ccrcc, ov, lscc, brca, gbm,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transcriptomics</th>\n",
       "      <td>[coad, ucec, pdac, ccrcc, ov, lscc, brca, gbm,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>broad</th>\n",
       "      <th>transcriptomics</th>\n",
       "      <td>[coad, ucec, pdac, ccrcc, ov, lscc, brca, gbm,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">harmonized</th>\n",
       "      <th>ancestry_prediction</th>\n",
       "      <td>[all_cancers]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>somatic_mutation</th>\n",
       "      <td>[all_cancers]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mssm</th>\n",
       "      <th>clinical</th>\n",
       "      <td>[all_cancers]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">umich</th>\n",
       "      <th>acetylproteomics</th>\n",
       "      <td>[ucec, lscc, brca, gbm, luad]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>phosphoproteomics</th>\n",
       "      <td>[coad, ucec, pdac, ccrcc, lscc, brca, gbm, lua...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>proteomics</th>\n",
       "      <td>[coad, ucec, pdac, ccrcc, ov, lscc, brca, gbm,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"10\" valign=\"top\">washu</th>\n",
       "      <th>CNV</th>\n",
       "      <td>[coad, ucec, pdac, ccrcc, ov, lscc, brca, gbm,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cibersort</th>\n",
       "      <td>[coad, ucec, pdac, ccrcc, ov, lscc, brca, gbm,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hla_typing</th>\n",
       "      <td>[all_cancers]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mature_miRNA</th>\n",
       "      <td>[ucec, pdac, ccrcc, lscc, gbm, luad, hnscc]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>precursor_miRNA</th>\n",
       "      <td>[ucec, pdac, ccrcc, lscc, gbm, luad, hnscc]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>somatic_mutation</th>\n",
       "      <td>[coad, ucec, pdac, ov, lscc, gbm, luad, hnscc]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>total_miRNA</th>\n",
       "      <td>[ucec, pdac, ccrcc, lscc, gbm, luad, hnscc]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transcriptomics</th>\n",
       "      <td>[coad, ucec, pdac, ccrcc, ov, brca, gbm, luad,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tumor_purity</th>\n",
       "      <td>[all_cancers]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>xcell</th>\n",
       "      <td>[coad, ucec, pdac, ccrcc, ov, lscc, brca, gbm,...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                           Cancer\n",
       "Source     Datatype                                                              \n",
       "bcm        CNV                  [coad, ucec, pdac, ccrcc, ov, lscc, brca, gbm,...\n",
       "           circular_RNA               [ucec, pdac, ccrcc, lscc, gbm, luad, hnscc]\n",
       "           miRNA                [coad, ucec, pdac, ccrcc, lscc, brca, gbm, lua...\n",
       "           phosphoproteomics    [coad, ucec, pdac, ccrcc, ov, lscc, brca, gbm,...\n",
       "           proteomics           [coad, ucec, pdac, ccrcc, ov, lscc, brca, gbm,...\n",
       "           transcriptomics      [coad, ucec, pdac, ccrcc, ov, lscc, brca, gbm,...\n",
       "broad      transcriptomics      [coad, ucec, pdac, ccrcc, ov, lscc, brca, gbm,...\n",
       "harmonized ancestry_prediction                                      [all_cancers]\n",
       "           somatic_mutation                                         [all_cancers]\n",
       "mssm       clinical                                                 [all_cancers]\n",
       "umich      acetylproteomics                         [ucec, lscc, brca, gbm, luad]\n",
       "           phosphoproteomics    [coad, ucec, pdac, ccrcc, lscc, brca, gbm, lua...\n",
       "           proteomics           [coad, ucec, pdac, ccrcc, ov, lscc, brca, gbm,...\n",
       "washu      CNV                  [coad, ucec, pdac, ccrcc, ov, lscc, brca, gbm,...\n",
       "           cibersort            [coad, ucec, pdac, ccrcc, ov, lscc, brca, gbm,...\n",
       "           hla_typing                                               [all_cancers]\n",
       "           mature_miRNA               [ucec, pdac, ccrcc, lscc, gbm, luad, hnscc]\n",
       "           precursor_miRNA            [ucec, pdac, ccrcc, lscc, gbm, luad, hnscc]\n",
       "           somatic_mutation        [coad, ucec, pdac, ov, lscc, gbm, luad, hnscc]\n",
       "           total_miRNA                [ucec, pdac, ccrcc, lscc, gbm, luad, hnscc]\n",
       "           transcriptomics      [coad, ucec, pdac, ccrcc, ov, brca, gbm, luad,...\n",
       "           tumor_purity                                             [all_cancers]\n",
       "           xcell                [coad, ucec, pdac, ccrcc, ov, lscc, brca, gbm,..."
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cptac.get_source_options()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Y queremos datos transcriptómicos, que como vemos tienen origen bcm, washu y broad. Creo que cada set tiene las mismas muestras, aunque diferente procesado, pero tengo que revisarlo."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "brca = cptac.Brca()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Data type</th>\n",
       "      <th>Available sources</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CNV</td>\n",
       "      <td>[bcm, washu]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>miRNA</td>\n",
       "      <td>[bcm]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>phosphoproteomics</td>\n",
       "      <td>[bcm, umich]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>proteomics</td>\n",
       "      <td>[bcm, umich]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>transcriptomics</td>\n",
       "      <td>[bcm, broad, washu]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ancestry_prediction</td>\n",
       "      <td>[harmonized]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>somatic_mutation</td>\n",
       "      <td>[harmonized, washu]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>clinical</td>\n",
       "      <td>[mssm]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>follow-up</td>\n",
       "      <td>[mssm]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>medical_history</td>\n",
       "      <td>[mssm]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>acetylproteomics</td>\n",
       "      <td>[umich]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>cibersort</td>\n",
       "      <td>[washu]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>hla_typing</td>\n",
       "      <td>[washu]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>tumor_purity</td>\n",
       "      <td>[washu]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>xcell</td>\n",
       "      <td>[washu]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              Data type    Available sources\n",
       "0                   CNV         [bcm, washu]\n",
       "1                 miRNA                [bcm]\n",
       "2     phosphoproteomics         [bcm, umich]\n",
       "3            proteomics         [bcm, umich]\n",
       "4       transcriptomics  [bcm, broad, washu]\n",
       "5   ancestry_prediction         [harmonized]\n",
       "6      somatic_mutation  [harmonized, washu]\n",
       "7              clinical               [mssm]\n",
       "8             follow-up               [mssm]\n",
       "9       medical_history               [mssm]\n",
       "10     acetylproteomics              [umich]\n",
       "11            cibersort              [washu]\n",
       "12           hla_typing              [washu]\n",
       "13         tumor_purity              [washu]\n",
       "14                xcell              [washu]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brca.list_data_sources()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "De los tipos de datos que incluye Brca, nos interesan \"transcriptomics\" y \"clinical\", en principio. \"follow-up\" y \"medical_history\" también pueden ser útiles."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Dataframe brca transcriptomics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr:last-of-type th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A1BG-AS1</th>\n",
       "      <th>A1CF</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2M-AS1</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A2ML1-AS1</th>\n",
       "      <th>A2ML1-AS2</th>\n",
       "      <th>A2MP1</th>\n",
       "      <th>A3GALT2</th>\n",
       "      <th>...</th>\n",
       "      <th>ZXDB</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11A</th>\n",
       "      <th>ZYG11AP1</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZYXP1</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>hsa-mir-1253</th>\n",
       "      <th>hsa-mir-423</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Database_ID</th>\n",
       "      <th>ENSG00000121410.12</th>\n",
       "      <th>ENSG00000268895.6</th>\n",
       "      <th>ENSG00000148584.15</th>\n",
       "      <th>ENSG00000175899.15</th>\n",
       "      <th>ENSG00000245105.4</th>\n",
       "      <th>ENSG00000166535.20</th>\n",
       "      <th>ENSG00000256661.1</th>\n",
       "      <th>ENSG00000256904.1</th>\n",
       "      <th>ENSG00000256069.7</th>\n",
       "      <th>ENSG00000184389.9</th>\n",
       "      <th>...</th>\n",
       "      <th>ENSG00000198455.4</th>\n",
       "      <th>ENSG00000070476.15</th>\n",
       "      <th>ENSG00000203995.10</th>\n",
       "      <th>ENSG00000232242.2</th>\n",
       "      <th>ENSG00000162378.13</th>\n",
       "      <th>ENSG00000159840.16</th>\n",
       "      <th>ENSG00000274572.1</th>\n",
       "      <th>ENSG00000074755.15</th>\n",
       "      <th>ENSG00000272920.1</th>\n",
       "      <th>ENSG00000266919.3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>01BR001</th>\n",
       "      <td>3.51</td>\n",
       "      <td>8.04</td>\n",
       "      <td>0.00</td>\n",
       "      <td>12.93</td>\n",
       "      <td>5.01</td>\n",
       "      <td>4.88</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.13</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>9.02</td>\n",
       "      <td>8.88</td>\n",
       "      <td>7.07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.86</td>\n",
       "      <td>10.63</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10.05</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01BR008</th>\n",
       "      <td>2.84</td>\n",
       "      <td>8.24</td>\n",
       "      <td>2.50</td>\n",
       "      <td>13.02</td>\n",
       "      <td>5.67</td>\n",
       "      <td>10.10</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>7.93</td>\n",
       "      <td>8.84</td>\n",
       "      <td>8.60</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8.44</td>\n",
       "      <td>11.36</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10.20</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01BR009</th>\n",
       "      <td>4.85</td>\n",
       "      <td>8.11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>13.68</td>\n",
       "      <td>5.87</td>\n",
       "      <td>9.50</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.11</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>7.53</td>\n",
       "      <td>8.85</td>\n",
       "      <td>3.64</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8.90</td>\n",
       "      <td>10.58</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10.54</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01BR010</th>\n",
       "      <td>4.24</td>\n",
       "      <td>7.96</td>\n",
       "      <td>0.00</td>\n",
       "      <td>12.64</td>\n",
       "      <td>4.82</td>\n",
       "      <td>5.03</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.33</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>8.53</td>\n",
       "      <td>8.31</td>\n",
       "      <td>8.28</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.28</td>\n",
       "      <td>12.40</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10.42</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01BR015</th>\n",
       "      <td>3.89</td>\n",
       "      <td>7.56</td>\n",
       "      <td>1.58</td>\n",
       "      <td>13.55</td>\n",
       "      <td>5.32</td>\n",
       "      <td>3.31</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.99</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>9.17</td>\n",
       "      <td>9.05</td>\n",
       "      <td>5.08</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10.13</td>\n",
       "      <td>10.44</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10.46</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 60591 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name                      A1BG          A1BG-AS1               A1CF  \\\n",
       "Database_ID ENSG00000121410.12 ENSG00000268895.6 ENSG00000148584.15   \n",
       "Patient_ID                                                            \n",
       "01BR001                   3.51              8.04               0.00   \n",
       "01BR008                   2.84              8.24               2.50   \n",
       "01BR009                   4.85              8.11               0.00   \n",
       "01BR010                   4.24              7.96               0.00   \n",
       "01BR015                   3.89              7.56               1.58   \n",
       "\n",
       "Name                       A2M           A2M-AS1              A2ML1  \\\n",
       "Database_ID ENSG00000175899.15 ENSG00000245105.4 ENSG00000166535.20   \n",
       "Patient_ID                                                            \n",
       "01BR001                  12.93              5.01               4.88   \n",
       "01BR008                  13.02              5.67              10.10   \n",
       "01BR009                  13.68              5.87               9.50   \n",
       "01BR010                  12.64              4.82               5.03   \n",
       "01BR015                  13.55              5.32               3.31   \n",
       "\n",
       "Name                A2ML1-AS1         A2ML1-AS2             A2MP1  \\\n",
       "Database_ID ENSG00000256661.1 ENSG00000256904.1 ENSG00000256069.7   \n",
       "Patient_ID                                                          \n",
       "01BR001                   0.0               0.0              3.13   \n",
       "01BR008                   0.0               0.0              0.00   \n",
       "01BR009                   0.0               0.0              3.11   \n",
       "01BR010                   0.0               0.0              1.33   \n",
       "01BR015                   0.0               0.0              2.99   \n",
       "\n",
       "Name                  A3GALT2  ...              ZXDB               ZXDC  \\\n",
       "Database_ID ENSG00000184389.9  ... ENSG00000198455.4 ENSG00000070476.15   \n",
       "Patient_ID                     ...                                        \n",
       "01BR001                   0.0  ...              9.02               8.88   \n",
       "01BR008                   0.0  ...              7.93               8.84   \n",
       "01BR009                   0.0  ...              7.53               8.85   \n",
       "01BR010                   0.0  ...              8.53               8.31   \n",
       "01BR015                   0.0  ...              9.17               9.05   \n",
       "\n",
       "Name                    ZYG11A          ZYG11AP1             ZYG11B  \\\n",
       "Database_ID ENSG00000203995.10 ENSG00000232242.2 ENSG00000162378.13   \n",
       "Patient_ID                                                            \n",
       "01BR001                   7.07               0.0               9.86   \n",
       "01BR008                   8.60               0.0               8.44   \n",
       "01BR009                   3.64               0.0               8.90   \n",
       "01BR010                   8.28               0.0               9.28   \n",
       "01BR015                   5.08               0.0              10.13   \n",
       "\n",
       "Name                       ZYX             ZYXP1              ZZEF1  \\\n",
       "Database_ID ENSG00000159840.16 ENSG00000274572.1 ENSG00000074755.15   \n",
       "Patient_ID                                                            \n",
       "01BR001                  10.63               0.0              10.05   \n",
       "01BR008                  11.36               0.0              10.20   \n",
       "01BR009                  10.58               0.0              10.54   \n",
       "01BR010                  12.40               0.0              10.42   \n",
       "01BR015                  10.44               0.0              10.46   \n",
       "\n",
       "Name             hsa-mir-1253       hsa-mir-423  \n",
       "Database_ID ENSG00000272920.1 ENSG00000266919.3  \n",
       "Patient_ID                                       \n",
       "01BR001                   0.0               0.0  \n",
       "01BR008                   0.0               0.0  \n",
       "01BR009                   0.0               0.0  \n",
       "01BR010                   0.0               0.0  \n",
       "01BR015                   0.0               0.0  \n",
       "\n",
       "[5 rows x 60591 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brca_transc_bcm = brca.get_transcriptomics('bcm')\n",
    "brca_transc_bcm.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr:last-of-type th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th colspan=\"5\" halign=\"left\">A1BG</th>\n",
       "      <th colspan=\"5\" halign=\"left\">A1BG-AS1</th>\n",
       "      <th>...</th>\n",
       "      <th colspan=\"8\" halign=\"left\">ZZEF1</th>\n",
       "      <th>hsa-mir-1253</th>\n",
       "      <th>hsa-mir-423</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>ENST00000263100.8</th>\n",
       "      <th>ENST00000595014.1</th>\n",
       "      <th>ENST00000596924.1</th>\n",
       "      <th>ENST00000598345.1</th>\n",
       "      <th>ENST00000600966.1</th>\n",
       "      <th>ENST00000593374.5</th>\n",
       "      <th>ENST00000593960.5</th>\n",
       "      <th>ENST00000594950.5</th>\n",
       "      <th>ENST00000595302.1</th>\n",
       "      <th>ENST00000599728.5</th>\n",
       "      <th>...</th>\n",
       "      <th>ENST00000572426.5</th>\n",
       "      <th>ENST00000572699.1</th>\n",
       "      <th>ENST00000572831.1</th>\n",
       "      <th>ENST00000573183.1</th>\n",
       "      <th>ENST00000573536.1</th>\n",
       "      <th>ENST00000573606.2</th>\n",
       "      <th>ENST00000574474.1</th>\n",
       "      <th>ENST00000575428.1</th>\n",
       "      <th>ENST00000609567.1</th>\n",
       "      <th>ENST00000586878.1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Database_ID</th>\n",
       "      <th>ENSG00000121410.12</th>\n",
       "      <th>ENSG00000121410.12</th>\n",
       "      <th>ENSG00000121410.12</th>\n",
       "      <th>ENSG00000121410.12</th>\n",
       "      <th>ENSG00000121410.12</th>\n",
       "      <th>ENSG00000268895.6</th>\n",
       "      <th>ENSG00000268895.6</th>\n",
       "      <th>ENSG00000268895.6</th>\n",
       "      <th>ENSG00000268895.6</th>\n",
       "      <th>ENSG00000268895.6</th>\n",
       "      <th>...</th>\n",
       "      <th>ENSG00000074755.15</th>\n",
       "      <th>ENSG00000074755.15</th>\n",
       "      <th>ENSG00000074755.15</th>\n",
       "      <th>ENSG00000074755.15</th>\n",
       "      <th>ENSG00000074755.15</th>\n",
       "      <th>ENSG00000074755.15</th>\n",
       "      <th>ENSG00000074755.15</th>\n",
       "      <th>ENSG00000074755.15</th>\n",
       "      <th>ENSG00000272920.1</th>\n",
       "      <th>ENSG00000266919.3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>01BR001</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.15</td>\n",
       "      <td>10.13</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.21</td>\n",
       "      <td>1.26</td>\n",
       "      <td>1.28</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.27</td>\n",
       "      <td>0.62</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.38</td>\n",
       "      <td>0.19</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01BR008</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.06</td>\n",
       "      <td>3.12</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2.98</td>\n",
       "      <td>1.94</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.12</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.74</td>\n",
       "      <td>0.27</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01BR009</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.47</td>\n",
       "      <td>11.88</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.22</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.32</td>\n",
       "      <td>1.19</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.53</td>\n",
       "      <td>1.55</td>\n",
       "      <td>0.32</td>\n",
       "      <td>0.28</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01BR010</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.16</td>\n",
       "      <td>0.25</td>\n",
       "      <td>12.34</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.41</td>\n",
       "      <td>2.05</td>\n",
       "      <td>0.79</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.30</td>\n",
       "      <td>1.66</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.60</td>\n",
       "      <td>1.21</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01BR015</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.39</td>\n",
       "      <td>19.99</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.93</td>\n",
       "      <td>0.79</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.77</td>\n",
       "      <td>1.29</td>\n",
       "      <td>0.48</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.08</td>\n",
       "      <td>0.50</td>\n",
       "      <td>0.36</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 225997 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name                        A1BG                                        \\\n",
       "Transcript_ID  ENST00000263100.8  ENST00000595014.1  ENST00000596924.1   \n",
       "Database_ID   ENSG00000121410.12 ENSG00000121410.12 ENSG00000121410.12   \n",
       "Patient_ID                                                               \n",
       "01BR001                      0.0               0.06               0.15   \n",
       "01BR008                      0.0               0.09               0.06   \n",
       "01BR009                      0.0               0.13               0.47   \n",
       "01BR010                      0.0               0.16               0.25   \n",
       "01BR015                      0.0               0.04               0.39   \n",
       "\n",
       "Name                                                         A1BG-AS1  \\\n",
       "Transcript_ID  ENST00000598345.1  ENST00000600966.1 ENST00000593374.5   \n",
       "Database_ID   ENSG00000121410.12 ENSG00000121410.12 ENSG00000268895.6   \n",
       "Patient_ID                                                              \n",
       "01BR001                    10.13               0.00               0.0   \n",
       "01BR008                     3.12               0.11               0.0   \n",
       "01BR009                    11.88               0.00               0.0   \n",
       "01BR010                    12.34               0.00               0.0   \n",
       "01BR015                    19.99               0.00               0.0   \n",
       "\n",
       "Name                                                                 \\\n",
       "Transcript_ID ENST00000593960.5 ENST00000594950.5 ENST00000595302.1   \n",
       "Database_ID   ENSG00000268895.6 ENSG00000268895.6 ENSG00000268895.6   \n",
       "Patient_ID                                                            \n",
       "01BR001                    0.21              1.26              1.28   \n",
       "01BR008                    0.00              2.98              1.94   \n",
       "01BR009                    0.00              1.22              0.10   \n",
       "01BR010                    0.41              2.05              0.79   \n",
       "01BR015                    0.00              1.93              0.79   \n",
       "\n",
       "Name                             ...              ZZEF1                     \\\n",
       "Transcript_ID ENST00000599728.5  ...  ENST00000572426.5  ENST00000572699.1   \n",
       "Database_ID   ENSG00000268895.6  ... ENSG00000074755.15 ENSG00000074755.15   \n",
       "Patient_ID                       ...                                         \n",
       "01BR001                     0.0  ...               0.27               0.62   \n",
       "01BR008                     0.0  ...               0.05               0.12   \n",
       "01BR009                     0.0  ...               0.32               1.19   \n",
       "01BR010                     0.0  ...               0.30               1.66   \n",
       "01BR015                     0.0  ...               0.77               1.29   \n",
       "\n",
       "Name                                                                    \\\n",
       "Transcript_ID  ENST00000572831.1  ENST00000573183.1  ENST00000573536.1   \n",
       "Database_ID   ENSG00000074755.15 ENSG00000074755.15 ENSG00000074755.15   \n",
       "Patient_ID                                                               \n",
       "01BR001                     0.00                0.0               1.38   \n",
       "01BR008                     0.00                0.0               0.74   \n",
       "01BR009                     0.00                0.0               3.53   \n",
       "01BR010                     0.00                0.0               1.60   \n",
       "01BR015                     0.48                0.0               2.08   \n",
       "\n",
       "Name                                                                    \\\n",
       "Transcript_ID  ENST00000573606.2  ENST00000574474.1  ENST00000575428.1   \n",
       "Database_ID   ENSG00000074755.15 ENSG00000074755.15 ENSG00000074755.15   \n",
       "Patient_ID                                                               \n",
       "01BR001                     0.19               0.14               0.00   \n",
       "01BR008                     0.27               0.03               0.00   \n",
       "01BR009                     1.55               0.32               0.28   \n",
       "01BR010                     1.21               0.10               0.00   \n",
       "01BR015                     0.50               0.36               0.00   \n",
       "\n",
       "Name               hsa-mir-1253       hsa-mir-423  \n",
       "Transcript_ID ENST00000609567.1 ENST00000586878.1  \n",
       "Database_ID   ENSG00000272920.1 ENSG00000266919.3  \n",
       "Patient_ID                                         \n",
       "01BR001                     0.0               0.0  \n",
       "01BR008                     0.0               0.0  \n",
       "01BR009                     0.0               0.0  \n",
       "01BR010                     0.0               0.0  \n",
       "01BR015                     0.0               0.0  \n",
       "\n",
       "[5 rows x 225997 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brca_transc_broad = brca.get_transcriptomics('broad')\n",
    "brca_transc_broad.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "La secuenciación del broad incluye diferentes isoformas o versiones del mismo gen. Por ejemplo, todas las columnas con Database_ID = ENSG00000121410.12 son isoformas del gen A1BG."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr:last-of-type th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th colspan=\"10\" halign=\"left\">5S_rRNA</th>\n",
       "      <th>...</th>\n",
       "      <th colspan=\"9\" halign=\"left\">uc_338</th>\n",
       "      <th>yR211F11.2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Database_ID</th>\n",
       "      <th>ENSG00000201285.1</th>\n",
       "      <th>ENSG00000212595.1</th>\n",
       "      <th>ENSG00000252830.2</th>\n",
       "      <th>ENSG00000271924.1</th>\n",
       "      <th>ENSG00000272253.1</th>\n",
       "      <th>ENSG00000272351.1</th>\n",
       "      <th>ENSG00000272435.1</th>\n",
       "      <th>ENSG00000274059.1</th>\n",
       "      <th>ENSG00000274097.1</th>\n",
       "      <th>ENSG00000274164.1</th>\n",
       "      <th>...</th>\n",
       "      <th>ENSG00000278040.1</th>\n",
       "      <th>ENSG00000278113.1</th>\n",
       "      <th>ENSG00000278218.1</th>\n",
       "      <th>ENSG00000278236.1</th>\n",
       "      <th>ENSG00000278333.1</th>\n",
       "      <th>ENSG00000278413.1</th>\n",
       "      <th>ENSG00000278498.1</th>\n",
       "      <th>ENSG00000278587.1</th>\n",
       "      <th>ENSG00000278643.1</th>\n",
       "      <th>ENSG00000213076.3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>01BR001</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.756644</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.238985</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01BR008</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.783371</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01BR009</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.732229</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.263943</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.156545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01BR010</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.023434</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.229401</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.242067</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.068029</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01BR015</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.248619</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 60483 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name                  5S_rRNA                                      \\\n",
       "Database_ID ENSG00000201285.1 ENSG00000212595.1 ENSG00000252830.2   \n",
       "Patient_ID                                                          \n",
       "01BR001                   0.0               0.0               0.0   \n",
       "01BR008                   0.0               0.0               0.0   \n",
       "01BR009                   0.0               0.0               0.0   \n",
       "01BR010                   0.0               0.0               0.0   \n",
       "01BR015                   0.0               0.0               0.0   \n",
       "\n",
       "Name                                                               \\\n",
       "Database_ID ENSG00000271924.1 ENSG00000272253.1 ENSG00000272351.1   \n",
       "Patient_ID                                                          \n",
       "01BR001              1.756644               0.0               0.0   \n",
       "01BR008              1.783371               0.0               0.0   \n",
       "01BR009              0.000000               0.0               0.0   \n",
       "01BR010              2.023434               0.0               0.0   \n",
       "01BR015              0.000000               0.0               0.0   \n",
       "\n",
       "Name                                                               \\\n",
       "Database_ID ENSG00000272435.1 ENSG00000274059.1 ENSG00000274097.1   \n",
       "Patient_ID                                                          \n",
       "01BR001                   0.0               0.0               0.0   \n",
       "01BR008                   0.0               0.0               0.0   \n",
       "01BR009                   0.0               0.0               0.0   \n",
       "01BR010                   0.0               0.0               0.0   \n",
       "01BR015                   0.0               0.0               0.0   \n",
       "\n",
       "Name                           ...            uc_338                    \\\n",
       "Database_ID ENSG00000274164.1  ... ENSG00000278040.1 ENSG00000278113.1   \n",
       "Patient_ID                     ...                                       \n",
       "01BR001              0.000000  ...               0.0               0.0   \n",
       "01BR008              0.000000  ...               0.0               0.0   \n",
       "01BR009              0.732229  ...               0.0               0.0   \n",
       "01BR010              0.000000  ...               0.0               0.0   \n",
       "01BR015              0.000000  ...               0.0               0.0   \n",
       "\n",
       "Name                                                               \\\n",
       "Database_ID ENSG00000278218.1 ENSG00000278236.1 ENSG00000278333.1   \n",
       "Patient_ID                                                          \n",
       "01BR001                   0.0          0.238985               0.0   \n",
       "01BR008                   0.0          0.000000               0.0   \n",
       "01BR009                   0.0          0.263943               0.0   \n",
       "01BR010                   0.0          0.229401               0.0   \n",
       "01BR015                   0.0          0.248619               0.0   \n",
       "\n",
       "Name                                                               \\\n",
       "Database_ID ENSG00000278413.1 ENSG00000278498.1 ENSG00000278587.1   \n",
       "Patient_ID                                                          \n",
       "01BR001                   0.0          0.000000               0.0   \n",
       "01BR008                   0.0          0.000000               0.0   \n",
       "01BR009                   0.0          0.000000               0.0   \n",
       "01BR010                   0.0          0.242067               0.0   \n",
       "01BR015                   0.0          0.000000               0.0   \n",
       "\n",
       "Name                                 yR211F11.2  \n",
       "Database_ID ENSG00000278643.1 ENSG00000213076.3  \n",
       "Patient_ID                                       \n",
       "01BR001                   0.0          0.000000  \n",
       "01BR008                   0.0          0.000000  \n",
       "01BR009                   0.0          0.156545  \n",
       "01BR010                   0.0          0.068029  \n",
       "01BR015                   0.0          0.000000  \n",
       "\n",
       "[5 rows x 60483 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brca_transc_washu = brca.get_transcriptomics('washu')\n",
    "brca_transc_washu.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "En el caso del database washu, no se incluyen distintas isoformas del mismo transcrito, y además muchas columnas no están asociadas a un nombre. La mejor opción parece ser el escoger el dataset bcm. Con el siguiente código, se descarga el df de interés: "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "brca_transc_bcm.to_csv(path_or_buf=\"data.tsv\", sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Dataframe brca clinical"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Downloading clinical_Pan-cancer.May2022.tsv.gz: 100%|████████████████████████████████| 243k/243k [00:00<00:00, 523kB/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>tumor_code</th>\n",
       "      <th>discovery_study</th>\n",
       "      <th>type_of_analyzed_samples</th>\n",
       "      <th>confirmatory_study</th>\n",
       "      <th>type_of_analyzed_samples</th>\n",
       "      <th>age</th>\n",
       "      <th>sex</th>\n",
       "      <th>race</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>ethnicity_race_ancestry_identified</th>\n",
       "      <th>...</th>\n",
       "      <th>additional_treatment_pharmaceutical_therapy_for_new_tumor</th>\n",
       "      <th>additional_treatment_immuno_for_new_tumor</th>\n",
       "      <th>number_of_days_from_date_of_initial_pathologic_diagnosis_to_date_of_additional_surgery_for_new_tumor_event_loco-regional</th>\n",
       "      <th>number_of_days_from_date_of_initial_pathologic_diagnosis_to_date_of_additional_surgery_for_new_tumor_event_metastasis</th>\n",
       "      <th>Recurrence-free survival, days</th>\n",
       "      <th>Recurrence-free survival from collection, days</th>\n",
       "      <th>Recurrence status (1, yes; 0, no)</th>\n",
       "      <th>Overall survival, days</th>\n",
       "      <th>Overall survival from collection, days</th>\n",
       "      <th>Survival status (1, dead; 0, alive)</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>01BR001</th>\n",
       "      <td>BR</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>55</td>\n",
       "      <td>Female</td>\n",
       "      <td>Black or African American</td>\n",
       "      <td>Not Hispanic or Latino</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>421.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01BR008</th>\n",
       "      <td>BR</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>48</td>\n",
       "      <td>Female</td>\n",
       "      <td>Black or African American</td>\n",
       "      <td>Not Hispanic or Latino</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01BR009</th>\n",
       "      <td>BR</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>64</td>\n",
       "      <td>Female</td>\n",
       "      <td>Black or African American</td>\n",
       "      <td>Not Hispanic or Latino</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01BR010</th>\n",
       "      <td>BR</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>65</td>\n",
       "      <td>Female</td>\n",
       "      <td>Black or African American</td>\n",
       "      <td>Not Hispanic or Latino</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01BR015</th>\n",
       "      <td>BR</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>35</td>\n",
       "      <td>Female</td>\n",
       "      <td>White</td>\n",
       "      <td>Not Hispanic or Latino</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>347.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 124 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name       tumor_code discovery_study type_of_analyzed_samples  \\\n",
       "Patient_ID                                                       \n",
       "01BR001            BR             Yes                      NaN   \n",
       "01BR008            BR             Yes                      NaN   \n",
       "01BR009            BR             Yes                      NaN   \n",
       "01BR010            BR             Yes                      NaN   \n",
       "01BR015            BR             Yes                      NaN   \n",
       "\n",
       "Name       confirmatory_study type_of_analyzed_samples age     sex  \\\n",
       "Patient_ID                                                           \n",
       "01BR001                   NaN                      NaN  55  Female   \n",
       "01BR008                   NaN                      NaN  48  Female   \n",
       "01BR009                   NaN                      NaN  64  Female   \n",
       "01BR010                   NaN                      NaN  65  Female   \n",
       "01BR015                   NaN                      NaN  35  Female   \n",
       "\n",
       "Name                             race               ethnicity  \\\n",
       "Patient_ID                                                      \n",
       "01BR001     Black or African American  Not Hispanic or Latino   \n",
       "01BR008     Black or African American  Not Hispanic or Latino   \n",
       "01BR009     Black or African American  Not Hispanic or Latino   \n",
       "01BR010     Black or African American  Not Hispanic or Latino   \n",
       "01BR015                         White  Not Hispanic or Latino   \n",
       "\n",
       "Name       ethnicity_race_ancestry_identified  ...  \\\n",
       "Patient_ID                                     ...   \n",
       "01BR001                                   NaN  ...   \n",
       "01BR008                                   NaN  ...   \n",
       "01BR009                                   NaN  ...   \n",
       "01BR010                                   NaN  ...   \n",
       "01BR015                                   NaN  ...   \n",
       "\n",
       "Name       additional_treatment_pharmaceutical_therapy_for_new_tumor  \\\n",
       "Patient_ID                                                             \n",
       "01BR001                                                   NaN          \n",
       "01BR008                                                   NaN          \n",
       "01BR009                                                   NaN          \n",
       "01BR010                                                   NaN          \n",
       "01BR015                                                   NaN          \n",
       "\n",
       "Name       additional_treatment_immuno_for_new_tumor  \\\n",
       "Patient_ID                                             \n",
       "01BR001                                          NaN   \n",
       "01BR008                                          NaN   \n",
       "01BR009                                          NaN   \n",
       "01BR010                                          NaN   \n",
       "01BR015                                          NaN   \n",
       "\n",
       "Name       number_of_days_from_date_of_initial_pathologic_diagnosis_to_date_of_additional_surgery_for_new_tumor_event_loco-regional  \\\n",
       "Patient_ID                                                                                                                            \n",
       "01BR001                                                   NaN                                                                         \n",
       "01BR008                                                   NaN                                                                         \n",
       "01BR009                                                   NaN                                                                         \n",
       "01BR010                                                   NaN                                                                         \n",
       "01BR015                                                   NaN                                                                         \n",
       "\n",
       "Name       number_of_days_from_date_of_initial_pathologic_diagnosis_to_date_of_additional_surgery_for_new_tumor_event_metastasis  \\\n",
       "Patient_ID                                                                                                                         \n",
       "01BR001                                                   NaN                                                                      \n",
       "01BR008                                                   NaN                                                                      \n",
       "01BR009                                                   NaN                                                                      \n",
       "01BR010                                                   NaN                                                                      \n",
       "01BR015                                                   NaN                                                                      \n",
       "\n",
       "Name       Recurrence-free survival, days  \\\n",
       "Patient_ID                                  \n",
       "01BR001                               NaN   \n",
       "01BR008                               NaN   \n",
       "01BR009                               NaN   \n",
       "01BR010                               NaN   \n",
       "01BR015                               NaN   \n",
       "\n",
       "Name       Recurrence-free survival from collection, days  \\\n",
       "Patient_ID                                                  \n",
       "01BR001                                               NaN   \n",
       "01BR008                                               NaN   \n",
       "01BR009                                               NaN   \n",
       "01BR010                                               NaN   \n",
       "01BR015                                               NaN   \n",
       "\n",
       "Name       Recurrence status (1, yes; 0, no) Overall survival, days  \\\n",
       "Patient_ID                                                            \n",
       "01BR001                                    0                  421.0   \n",
       "01BR008                                    0                    NaN   \n",
       "01BR009                                    0                    NaN   \n",
       "01BR010                                    0                    NaN   \n",
       "01BR015                                    0                  347.0   \n",
       "\n",
       "Name       Overall survival from collection, days  \\\n",
       "Patient_ID                                          \n",
       "01BR001                                       NaN   \n",
       "01BR008                                       NaN   \n",
       "01BR009                                       NaN   \n",
       "01BR010                                       NaN   \n",
       "01BR015                                       NaN   \n",
       "\n",
       "Name       Survival status (1, dead; 0, alive)  \n",
       "Patient_ID                                      \n",
       "01BR001                                    0.0  \n",
       "01BR008                                    NaN  \n",
       "01BR009                                    NaN  \n",
       "01BR010                                    NaN  \n",
       "01BR015                                    0.0  \n",
       "\n",
       "[5 rows x 124 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical = brca.get_clinical('mssm')\n",
    "clinical.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "De nuevo, descargamos los datos de interés con:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "clinical.to_csv(path_or_buf=\"clinical.tsv\", sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
